### PR TITLE
DashboardScene: Panel menu tracking, adding explore menu action and unit tests

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Adds tracking
* [x] Adds explore action 
* [x] Adds unit tests

### CODE REVIEW

File: packages/grafana-data/src/types/datasource.ts [hunk 1]
```diff
@@ -339,7 +339,7 @@ abstract class DataSourceApi<
 
   getVersion?(optionalOptions?: any): Promise<string>;
 
-  interpolateVariablesInQueries?(queries: TQuery[], scopedVars: ScopedVars | {}): TQuery[];
+  interpolateVariablesInQueries?(queries: TQuery[], scopedVars: ScopedVars): TQuery[];
```
Comment (torkelo) [inline packages/grafana-data/src/types/datasource.ts:342]: this `| {}` typing is unnecessary, ScopedVars already supports this (ie can be an empty object)

File: public/app/features/dashboard-scene/utils/utils.ts [hunk 2]
```diff
@@ -113,3 +120,19 @@ export function getMultiVariableValues(variable: MultiValueVariable) {
     texts: Array.isArray(text) ? text : [text],
   };
 }
+
+export function getQueryRunnerFor(sceneObject: SceneObject | undefined): SceneQueryRunner | undefined {
```
Comment (dprokop) [inline public/app/features/dashboard-scene/utils/utils.ts:124]: I see this function in 3rd pr, not sure if this is the same one ;)

**Review by dprokop — APPROVED** [review]

File: public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx [hunk 3]
```diff
@@ -1,60 +1,81 @@
...
+        text: t('panel.header-menu.explore', `Explore`),
+        iconClassName: 'compass',
+        shortcut: 'p x',
+        onClick: () => reportInteraction('dashboards_panelheader_menu', { item: 'explore' }),
+        href: await getExploreUrl({
+          queries: queryRunner.state.queries,
+          dsRef: queryRunner.state.datasource,
+          timeRange: timeRange.state.value,
+          scopedVars: { __sceneObject: { value: panel } },
+        }),
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx:65]: @dprokop yes, all menu items are generated when you open the menu

File: public/app/features/dashboard-scene/utils/utils.ts (hunk 2 above)
Comment (torkelo) [reply public/app/features/dashboard-scene/utils/utils.ts:124]: @dprokop I needed it in multiple PRs so will have to deal with the merge conflicts

**Pull request merged**